core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -488,9 +493,13 @@ private RexNode simplifyMultiply(RexCall e) {
   }
 
   private RexNode simplifyDivide(RexCall e) {
-    final int oneIndex = findLiteralIndex(e.operands, BigDecimal.ONE);
-    if (oneIndex == 1) {
-      RexNode leftOperand = e.getOperands().get(0);
+    RexNode leftOperand = e.getOperands().get(0);
+    RexNode rightOperand = e.getOperands().get(1);
+    if ((isSafeExpression(leftOperand) && STRONG.isNull(leftOperand))
```

Comment:
is the 'isSafeExpression' necessary?
If the value is NULL, isn't it always safe?

Reply:
This appears to be somewhat "new" simplification logic that was not there before. I get the feeling that simplifications of arithmetic operations with nulls is a problem that we should tackle in a more general fashion and not only for the division operator. From my point of view this deserves a separate JIRA ticket and PR dedicated to simplification of arithmetic operations with nulls. I see there is already some code doing this for comparison operators (`simplifyComparisonWithNull`).